tests/util_test.go

```diff
@@ -55,8 +57,68 @@ func (c *ClientTests) SetupTest() {
 		return cassette.DefaultMatcher(r, i) && (b.String() == "" || b.String() == i.Body)
 	})
 
-	r.AddFilter(func(i *cassette.Interaction) error {
-		delete(i.Request.Headers, "Authorization")
+	// Filter sensitive data from cassettes
+	// Replace value has to be type specific to its corresponding struct
+	responseBodyScrubbers := map[string]interface{}{
+		"api_keys":         []string{},
+		"children":         []string{},
+		"client_ip":        "REDACTED",
+		"test_credentials": map[string]string{},
+		"credentials":      map[string]string{},
+		"email":            "REDACTED",
+		"keys":             []*easypost.APIKey{},
+		"key":              "REDACTED",
+		"phone_number":     "REDACTED",
+		"phone":            "REDACTED",
+	}
+	r.AddSaveFilter(func(i *cassette.Interaction) error {
+		// Filter headers
+		i.Request.Headers["Authorization"] = []string{"REDACTED"}
+		i.Request.Headers["User-Agent"] = []string{"REDACTED"}
+		i.Request.Headers["X-Client-User-Agent"] = []string{"REDACTED"}
+
+		// Filter response bodies
+		var responseBodyByte []byte
+		var responseBodyString string
+
+		for scrubber, replaceValue := range responseBodyScrubbers {
+			if strings.Contains(i.Response.Body, scrubber) {
+				var responseBody = i.Response.Body
+
+				if responseBody[0] == '[' && responseBody[len(responseBody)-1] == ']' {
+					var arrayObjects []map[string]interface{}
+
+					err := json.Unmarshal([]byte(responseBody), &arrayObjects)
+					if err != nil {
+						panic(err)
```

Comment:
Let's use `os.Exit` here instead of `panic`: https://stackoverflow.com/a/28473339/6064135 (make sure to print the error message prior to using os.Exit.)

This goes for all the panics in this PR (you'll notice we don't use Panic elsewhere in the lib).

Reply:
resolved